From 08eb2411d7516886cdefd26303c564d9dbaad58f Mon Sep 17 00:00:00 2001
From: Robert Marko <robimarko@gmail.com>
Date: Tue, 8 Jun 2021 22:14:34 +0200
Subject: [PATCH 2/2] nss-crypto: replace ioremap_nocache() with ioremap

ioremap_nocache() was dropped in kernel 5.5 as regular
ioremap() was exactly the same.

So, simply replace all of the ioremap_nocache() calls
with ioremap().

Signed-off-by: Robert Marko <robimarko@gmail.com>
---
 v2.0/src/hal/ipq50xx/nss_crypto_ce5.c    | 4 ++--
 v2.0/src/hal/ipq60xx/nss_crypto_eip197.c | 2 +-
 v2.0/src/hal/ipq807x/nss_crypto_eip197.c | 2 +-
 3 files changed, 4 insertions(+), 4 deletions(-)

--- a/v2.0/src/hal/ipq50xx/nss_crypto_ce5.c
+++ b/v2.0/src/hal/ipq50xx/nss_crypto_ce5.c
@@ -288,7 +288,7 @@ int nss_crypto_ce5_engine_init(struct pl
 	 * remap the I/O addresses for crypto
 	 */
 	eng->crypto_paddr = crypto_res->start;
-	eng->crypto_vaddr = ioremap_nocache(crypto_res->start, resource_size(crypto_res));
+	eng->crypto_vaddr = ioremap(crypto_res->start, resource_size(crypto_res));
 	if (!eng->crypto_vaddr) {
 		nss_crypto_warn("%px: unable to remap crypto_addr(0x%px)\n", node, (void *)eng->crypto_paddr);
 		nss_crypto_engine_free(eng);
@@ -299,7 +299,7 @@ int nss_crypto_ce5_engine_init(struct pl
 	 * remap the I/O addresses for bam
 	 */
 	eng->dma_paddr = bam_res->start;
-	eng->dma_vaddr = ioremap_nocache(bam_res->start, resource_size(bam_res));
+	eng->dma_vaddr = ioremap(bam_res->start, resource_size(bam_res));
 	if (!eng->dma_vaddr) {
 		iounmap(eng->crypto_vaddr);
 		nss_crypto_warn("%px: unable to remap dma_addr(0x%px)\n", node, (void *)eng->dma_paddr);
--- a/v2.0/src/hal/ipq60xx/nss_crypto_eip197.c
+++ b/v2.0/src/hal/ipq60xx/nss_crypto_eip197.c
@@ -490,7 +490,7 @@ int nss_crypto_eip197_engine_init(struct
 	 * remap the I/O addresses
 	 */
 	paddr = res->start + offset;
-	vaddr = ioremap_nocache(paddr, resource_size(res));
+	vaddr = ioremap(paddr, resource_size(res));
 	if (!vaddr) {
 		nss_crypto_warn("%px: unable to remap crypto_addr(0x%px)\n", node, (void *)paddr);
 		return -EIO;
--- a/v2.0/src/hal/ipq807x/nss_crypto_eip197.c
+++ b/v2.0/src/hal/ipq807x/nss_crypto_eip197.c
@@ -490,7 +490,7 @@ int nss_crypto_eip197_engine_init(struct
 	 * remap the I/O addresses
 	 */
 	paddr = res->start + offset;
-	vaddr = ioremap_nocache(paddr, resource_size(res));
+	vaddr = ioremap(paddr, resource_size(res));
 	if (!vaddr) {
 		nss_crypto_warn("%px: unable to remap crypto_addr(0x%px)\n", node, (void *)paddr);
 		return -EIO;
